gtdynamics/utils/Trajectory.h

```diff
@@ -57,8 +58,22 @@ class Trajectory {
    * @param walk_cycle  The Walk Cycle for the robot.
    * @param repeat      The number of repetitions for each phase of the gait.
    */
-  Trajectory(const WalkCycle &walk_cycle, size_t repeat)
-      : repeat_(repeat), walk_cycle_(walk_cycle) {}
+  Trajectory(const WalkCycle &walk_cycle, size_t repeat) {
+    size_t k = 0;
+    // Loop over `repeat` walk cycles W_i
+    for (size_t i = 0; i < repeat; i++) {
+      // Creating the phases for the ith walk cycle, and append them phases_.
+      auto phases_i = walk_cycle.phases();
+      phases_.insert(phases_.end(), phases_i.begin(), phases_i.end());
+    }
+
+    for (auto&& phase : phases_) {
```

Comment:
document. But, also, should it be only done for walk_cycle.phases() ?

Reply:
contact_points_ was a member of WalkCycle and I moved it to trajectory because we took out walk_cycle_ as a member from Trajectory (also did this for other members functions from WalkCycle).
To build the contact_points_ vector now I Take from each phase in the trajectory the unique contact points and add them them to contact_points_. Is there a reason to do this only for the walk_cycle.phases()? These contact_points_ should be unique (that's what was done in WalkCycle as well).